examples-gallery/plot_crane_moving_a_load.py

```diff
@@ -0,0 +1,338 @@
+
+# %%
+# Animate the Simulation.
+#------------------------
+h_sol = solution[-1]
+fps = 20
+
+tf = h_sol*(num_nodes - 1)
+state_vals, input_vals, _ = parse_free(solution, len(state_symbols),
+                                       len(specified_symbols), num_nodes)
+t_arr = np.linspace(t0, tf, num_nodes)
+state_sol = CubicSpline(t_arr, state_vals.T)
+input_sol = CubicSpline(t_arr, input_vals.T)
+
+coordinates = P1.pos_from(O).to_matrix(N)
+coordinates = coordinates.row_join(P2.pos_from(O).to_matrix(N))
+
+pl, pl_vals = zip(*par_map.items())
+coords_lam = sm.lambdify((*state_symbols, F, *pl), coordinates, cse=True)
+
+width, height, radius = 0.5, 0.5, 0.5
+
+
+def init_plot():
+    xmin, xmax = starting_location - 1, ending_location + 1
+    ymin, ymax = -par_map[l] - 1, 1
+
+    fig, ax = plt.subplots(figsize=(8, 8))
+    ax.set_xlim(xmin, xmax)
+    ax.set_ylim(ymin, ymax)
+    ax.set_aspect('equal')
+    ax.set_xlabel('X-axis [m]')
+    ax.set_ylabel('Y-axis [m]')
+
+    ax.axhline(0., color='black', lw=1.5)
+    ax.axvline(ending_location+radius, color='black', lw=1.0)
+    ax.axvline(starting_location-radius, color='black', lw=1.0)
+    ax.annotate('walls', xy=(ending_location+radius, -2),
+                arrowprops=dict(arrowstyle='->',
+                                connectionstyle='arc3, rad=-.2',
+                                lw=0.25),
+                xytext=(8, 0.5), fontsize=9)
+    ax.annotate('', xy=(starting_location-radius, -1),
+                arrowprops=dict(arrowstyle='->',
+                                connectionstyle='arc3, rad=-.2',
+                                lw=0.25),
+                xytext=(8, 0.5))
+
+    line1, = ax.plot([], [], color='orange', lw=1)
+    pfeil = ax.quiver([], [], [], [], color='red', scale=55, width=0.002,
+                      headwidth=8)
+    recht = patches.Rectangle((-width/2, -height/2), width=width,
+                              height=height, fill=True, color='red',
+                              ec='black')
+    ax.add_patch(recht)
+    load = patches.Circle((0, -par_map[l]), radius=radius, fill=True,
+                          color='blue', ec='black')
+    ax.add_patch(load)
+
+    return fig, ax, line1, recht, load, pfeil
+
+
+fig, ax, line1, recht, load, pfeil = init_plot()
+
+
+def update(t):
+    message = f'running time {t:0.2f} sec \n The red arrow shows the force.'
+    ax.set_title(message, fontsize=12)
+
+    coords = coords_lam(*state_sol(t), input_sol(t), *pl_vals)
+
+    line1.set_data([coords[0, 0], coords[0, 1]], [coords[1, 0], coords[1, 1]])
+    recht.set_xy((coords[0, 0] - width/2., coords[1, 0] - height/2.))
+    load.set_center((coords[0, 1], coords[1, 1]))
+    pfeil.set_offsets([coords[0, 0], coords[1, 0]+0.25])
+    pfeil.set_UVC(input_sol(t), 0.25)
+    return line1, recht, load, pfeil
+
+anim = animation.FuncAnimation(fig, update,
+                               frames=np.arange(t0, tf, 1/fps),
+                               interval=1/fps*1000)
```

Comment:
The animation does not run until it hits the for me. Maybe it is running out of memory. I suggest reduce the fps and number of frames.